aten/src/ATen/native/cuda/CuFFTPlanCache.h

```diff
@@ -85,21 +224,26 @@ class CuFFTConfig {
   CuFFTConfig(const CuFFTConfig&) = delete;
   CuFFTConfig& operator=(CuFFTConfig const&) = delete;
 
-  explicit CuFFTConfig(Tensor& input, int64_t signal_ndim, bool complex_input,
-    bool complex_output, IntArrayRef checked_signal_sizes, bool onesided,
-    IntArrayRef output_sizes) {
+  explicit CuFFTConfig(const CuFFTParams& params):
+      CuFFTConfig(
+          IntArrayRef(params.input_strides_, params.signal_ndim_ + 1),
+          IntArrayRef(params.output_strides_, params.signal_ndim_ + 1),
+          IntArrayRef(params.sizes_, params.signal_ndim_ + 1),
+          params.fft_type_,
+          params.value_type_) {}
 
-    // signal sizes
-#ifdef __HIP_PLATFORM_HCC__
-    std::vector<int> signal_sizes(checked_signal_sizes.begin(),
-                                  checked_signal_sizes.end());
-#else
-    std::vector<long long int> signal_sizes(checked_signal_sizes.begin(),
-                                            checked_signal_sizes.end());
-#endif
+  // For complex types, strides are in units of 2 * element_size(dtype)
+  // sizes are for the full signal, including batch size and always two-sided
+  CuFFTConfig(IntArrayRef in_strides, IntArrayRef out_strides,
+      IntArrayRef sizes, CuFFTTransformType fft_type, ScalarType dtype):
+        fft_type_(fft_type), value_type_(dtype) {
+
+    // signal sizes (excluding batch dim)
+    CuFFTDimVector signal_sizes(sizes.begin() + 1, sizes.end());
 
     // input batch size
-    long long int batch = input.size(0);
+    long long int batch = sizes[0];
```

Comment:
Can we replace the long long int with an explicit type? I realize this is not an issue introduced with your PR, but it'd be nice to cleanup while we're here.